### PR TITLE
fix: QA of newly added docstrings.

### PR DESCRIPTION
**Pull request by shani-leroux**
### Description

- [ ] I've included a concise description of what each node does

### Styleguide

- [ ] My node adheres to the [styleguide](https://github.com/flojoy-io/nodes/blob/main/node_styleguide.md) for Flojoy nodes

### Docs

- [ ] I've submitted a PR for a documentation page for the new node(s) that contains usage examples (see docs.flojoy.io)

### Testing

- [ ] This PR includes a unit test ([example here](https://github.com/flojoy-io/nodes/blob/c28808f2d0aec8f116cdb3fccb46c4f5256574e9/TRANSFORMERS/ARITHMETIC/ADD/ADD_test_.py) and/or ideally a screenshot of the node's output on an example app.

### CODE REVIEW

File: LOADERS/LOCAL_FILE_SYSTEM/BATCH_PROCESSOR/BATCH_PROCESSOR.py [hunk 1]
```diff
@@ -22,41 +22,39 @@ def BATCH_PROCESSOR(
...
         A switching parameter that refreshes the cache of files. If a separate
         programme is expected to continuously write new files of interest to the
         directory, this flag will enable the update of the new files for processing.
 
     Returns
     -------
-    retval                  :       TextBlob
-        A `TextBlob` data container to pass to other nodes for processing
-
+    retval : TextBlob
```
Comment (JeffDotPng) [inline LOADERS/LOCAL_FILE_SYSTEM/BATCH_PROCESSOR/BATCH_PROCESSOR.py:56]: The name of this output should be `fname`, and there should be another return value here which is `n_files`

File: LOGIC_GATES/LOOP_TOOLS/BREAK/BREAK.py [hunk 2]
```diff
@@ -9,21 +9,21 @@ def BREAK(
...
-    referred_node       :       NodeReference
+    referred_node : NodeReference
         This is the specific instance of a LOOP node that you want to break.
         It is required to differentiate between multiple LOOPs if they exist
-        in the same application
-    default             :       Optional[DataContainer]
-        This can be whatever, this node simply requires an input (most likely from
-        the return of a CONDITIONAL node) so that it is executed
+        in the same application.
+    default : Optional [DataContainer]
```
Comment (JeffDotPng) [inline LOGIC_GATES/LOOP_TOOLS/BREAK/BREAK.py:22]: The type here should simply be `DataContainer`

File: TRANSFORMERS/IMAGE_PROCESSING/REGION_PROPERTIES/REGION_PROPERTIES.py [hunk 3]
```diff
@@ -14,46 +14,42 @@
...
+    - The contour centroid.
+    - The countour bounding-box.
+    - A mouse-hover utility that displays the contour information to the user.
 
     Parameters
     ----------
-    default     :       Optional[Image | Grayscale],
-        The input node to this function. If nothing is supplied, then a demo test
-        case is returned to illustrate the functionality of this node.
+    default : Optional [Image | Grayscale]
```
Comment (JeffDotPng) [inline TRANSFORMERS/IMAGE_PROCESSING/REGION_PROPERTIES/REGION_PROPERTIES.py:44]: Type here should be Image | Grayscale

File: TRANSFORMERS/IMAGE_PROCESSING/EXTREMA_DETERMINATION/EXTREMA_DETERMINATION.py [hunk 4]
```diff
@@ -40,142 +40,109 @@ def EXTREMA_DETERMINATION(
...
-        This object provides a mask to apply to the peak finding routines. Peaks
-        found by any algorithm inside this mask are ignored. Should be of a
-        datatype that can be static cast to booleans. If none, it is assumed
-        that the entire image is valid for peak detection.
-    center          :       list[int] , None
+    default : Image | Grayscale | Matrix
+        The input DataContainer that contains the image to be processed.
+        Can either be RGBA, greyscale, or a matrix type. 
+        In the case of RGB(A), the image is flattened to grayscale for the peak detection.
+    image_mask : Optional [Grayscale | Matrix], None
```
Comment (JeffDotPng) [inline TRANSFORMERS/IMAGE_PROCESSING/EXTREMA_DETERMINATION/EXTREMA_DETERMINATION.py:104]: Grayscale | Matrix

**Review by JeffDotPng — COMMENTED** [review]
Looks pretty good, nice work, just a few small changes needed
I don't have the background to review the changes to the docstring for `EXTREMA_DETERMINATION` so I will let @trbritt or @dstrande look at it.

**Comment by dstrande** [comment]
Might wait for @trbritt to see what he thinks. @JeffDotPng @itsjoeoui do you have an opinion on how complicated a docstring can be? 

Personally I think if a node has a complex use-case then it's fine if the docstring is very in depth.

**Comment by trbritt** [comment]
There were many reasons why I've kept these docstrings as in-depth as I did. 

The techniques they implement are complex and use latest research, and the first algorithm in the `EXTREMA_DETERMINATION` node was jointly developed by myself and a collaborator, which is why I give them credit in the doc string of that node. IMHO,  nodes that implement complicated algorithms should be allowed to keep a references section with an in-depth doc string, so that interested users have all the tools available to them to understand the functionality of the node.

All the changes look fine with the caveat of reincluding the References sections of the docstrings.

**Comment by JeffDotPng** [comment]
> There were many reasons why I've kept these docstrings as in-depth as I did.
> 
> The techniques they implement are complex and use latest research, and the first algorithm in the `EXTREMA_DETERMINATION` node was jointly developed by myself and a collaborator, which is why I give them credit in the doc string of that node. IMHO, nodes that implement complicated algorithms should be allowed to keep a references section with an in-depth doc string, so that interested users have all the tools available to them to understand the functionality of the node.
> 
> All the changes look fine with the caveat of reincluding the References sections of the docstrings.

I agree, plus users won't see this information unless they go on the docs or view the python source code in `studio` (no walls of text in the parameter descriptions) so it's not disruptive to UX.
The references section should definitely be reincluded!

File: TRANSFORMERS/IMAGE_PROCESSING/REGION_PROPERTIES/REGION_PROPERTIES.py [hunk 5]
```diff
@@ -14,46 +14,42 @@
...
-        - First, the regions of the INTEGER image are identified and labelled (NOTA BENE
-            the integer type requirement)
-        - Second, the regions are then analysed.
-    The first step is provided by the morphology library of scikit-image's label function,
-    while the second is provided by scikit-image's regionprops function.
+    - First, the regions of the INTEGER image are identified and labelled.
+    - Second, the regions are analysed.
+    
+    The first step is provided by the morphology library label function of the 'scikit-image',
+    while the second is provided by the regionprops function of the 'scikit-image'.
```
Comment (JeffDotPng) [inline TRANSFORMERS/IMAGE_PROCESSING/REGION_PROPERTIES/REGION_PROPERTIES.py:32]: The text from before is more accurate:
```
    The first step is provided by the morphology library of scikit-image's label function,
    while the second is provided by scikit-image's regionprops function.
```

File: TRANSFORMERS/IMAGE_PROCESSING/REGION_PROPERTIES/REGION_PROPERTIES.py [hunk 6]
```diff
@@ -14,46 +14,42 @@
...
+        The input node to this function. 
+        If nothing is supplied, a demo test case is returned to illustrate the functionality of this node.
 
     Returns
     -------
-    fig         :       Plotly
-        Returns the Plotly superclass of `DataContainer` that has the illustrated
+    fig : Plotly
+        Returns the Plotly superclass of 'DataContainer' that has the illustrated
         features as determined by this node.
```
Comment (JeffDotPng) [inline TRANSFORMERS/IMAGE_PROCESSING/REGION_PROPERTIES/REGION_PROPERTIES.py:52]:
```suggestion
        A Plotly figure containing the illustrated features as determined by this node.
```

**Review by JeffDotPng — CHANGES_REQUESTED** [review]
Noticed a few more things

**Comment by trbritt** [comment]
All of @JeffDotPng's proposed changes I approve for the record

**Review by JeffDotPng — APPROVED** [review]
lgtm

**Pull request merged**